### PR TITLE
fix: side menu icon should not be visible in mobile when there is no menu 

### PR DESCRIPTION
**Pull request by maze-runnar**
fixes : #3656 
Screenshot from 2019-11-26 20-18-59
**visible for other pages**
Screenshot from 2019-11-26 20-24-47

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The acceptance, integration, unit tests and linter pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: app/templates/public.hbs [hunk 1]
```diff
@@ -26,11 +26,13 @@
     <div class="ui stackable grid">
       <div class="three wide column" id="public-event-content">
         {{#if device.isMobile}}
-          <button {{action 'toggleMenu'}} class="ui basic icon button mobile only">
-            <i class="content icon"></i>
-          </button>
-          {{#if isMenuOpen}}
-            {{public/side-menu class='toggle menu' event=model}}
+          {{#if (and (not-eq session.currentRouteName 'public.cfs.new-session') (not-eq session.currentRouteName 'public.cfs.new-speaker') (not-eq session.currentRouteName 'public.cfs.edit-speaker') (not-eq session.currentRouteName 'public.cfs.edit-session'))}}
```
Comment (kushthedude) [inline app/templates/public.hbs]: Rather than using such conditionals in the templates, please use a computed property from controllers.

Comment (maze-runnar) [reply app/templates/public.hbs]: The same condition is used in side-menu.hbs file for visibility of menu .

Comment (iamareebjamal) [reply app/templates/public.hbs]: This is really brittle

Comment (kushthedude) [reply app/templates/public.hbs]: I suspected so, @maze-runnar make a computed property and use it for the above condition too

Comment (maze-runnar) [reply app/templates/public.hbs]: i  have made the computed property and it's working fine . may i make changes here ?

**Comment by maze-runnar** [comment]
@kushthedude please review

**Comment by niranjan94** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- app/controllers/public.js  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/fossasia/open-event-frontend/pullRequest?prid=4569334&bid=15360365)

**Comment by niranjan94** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- app/controllers/public.js  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/fossasia/open-event-frontend/pullRequest?prid=4569334&bid=15360365)

File: app/controllers/public.js [hunk 2]
```diff
@@ -11,6 +11,11 @@ export default Controller.extend({
   displayEndDate: computed('model.startsAtDate', 'model.endsAtDate', function() {
     return !moment(this.model.startsAtDate).isSame(this.model.endsAtDate, 'minute');
   }),
+  displaySideMenu: computed('session.currentRouteName', function() {
+    if (this.get('session.currentRouteName')) {
```
Comment (kushthedude) [inline app/controllers/public.js]: Dont use ES5 getter.

Comment (maze-runnar) [reply app/controllers/public.js]: then what ??

Comment (maze-runnar) [reply app/controllers/public.js]: should i migrate it to ES6?

Comment (kushthedude) [reply app/controllers/public.js]: No, ~this.get('session')~ this.session

Comment (maze-runnar) [reply app/controllers/public.js]: in the same file :
```
smallLead: computed('session.currentRouteName', function() {
    if (this.get('session.currentRouteName')) {
      return this.get('session.currentRouteName') !== 'public.index';
    }
  }),
```
Should i change  ES5 here too ?

Comment (maze-runnar) [reply app/controllers/public.js]: in the same file :

```
smallLead: computed('session.currentRouteName', function() {
    if (this.get('session.currentRouteName')) {
      return this.get('session.currentRouteName') !== 'public.index';
    }
  }),
```
Should i change ES5 here too ?

**Comment by niranjan94** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- app/controllers/public.js  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/fossasia/open-event-frontend/pullRequest?prid=4569334&bid=15360365)

**Comment by maze-runnar** [comment]
@kushthedude  should i open a seperate PR for migrating  "app/controllers/public.js" to ES6.
As you have said migrate those on whom currently working during resolving that issue.

**Comment by niranjan94** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- app/controllers/public.js  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/fossasia/open-event-frontend/pullRequest?prid=4569334&bid=15360365)

**Comment by maze-runnar** [comment]
screenshot after above change .
Screenshot from 2019-12-22 16-33-46

**Pull request merged**